### PR TITLE
test: codspeed CI workflow

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [x] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #805 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

- I am opening this with 3 queries only for the sake of example
- I wonder if the data size we are using is relevant

### CODE REVIEW

File: tpch/benchmarks/queries_test.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    nation = nw.from_native(read_fn(DATA_FOLDER / "nation.parquet")).lazy()
+    orders = nw.from_native(read_fn(DATA_FOLDER / "orders.parquet")).lazy()
+    part = nw.from_native(read_fn(DATA_FOLDER / "part.parquet")).lazy()
+    partsupp = nw.from_native(read_fn(DATA_FOLDER / "partsupp.parquet")).lazy()
+    region = nw.from_native(read_fn(DATA_FOLDER / "region.parquet")).lazy()
+    supplier = nw.from_native(read_fn(DATA_FOLDER / "supplier.parquet")).lazy()
+
+    _ = benchmark(q1, lineitem)
+    _ = benchmark(q2, region, nation, supplier, part, partsupp)
+    _ = benchmark(q3, customer, lineitem, orders)
```
Comment (FBruzzesi) [inline tpch/benchmarks/queries_test.py:35]: I am not sure that having multiple benchmarks in a single test is ok 🤔

Comment (DeaMariaLeon) [reply tpch/benchmarks/queries_test.py:35]: You can use parametrize, yes, but you need to create different names for each result.

**Comment by DeaMariaLeon** [comment]
Maybe you did it, but just in case: Don't forget to add Narwhals to CodSpeed (you need admin rights). Even if you were able to previously test your fork.

**Comment by FBruzzesi** [comment]
> Maybe you did it, but just in case: Don't forget to add Narwhals to CodSpeed (you need admin rights). Even if you were able to previously test your fork.

Maybe I am missing something, but checking in [repo settings](https://github.com/narwhals-dev/narwhals/settings/installations), I am able to see it enabled. Adding screenshots:

image

And then in _configure_:

image

From the error message it seems we still need a token?

**Comment by DeaMariaLeon** [comment]
I don't have access to: [repo settings](https://github.com/narwhals-dev/narwhals/settings/installations)

OK. I didn't need a token. Plus it says that the repo has not been enabled. 
What do you see on Narwhals permissions on Codspeed? BTW Codspeed only accepts organizations.

**Comment by DeaMariaLeon** [comment]
Your name must be listed as admin or provider admin on "permissions":

Screenshot 2024-08-20 at 09 35 56

**Comment by FBruzzesi** [comment]
I should have enabled it, I will make a commit to trigger all the workflows one more time :)

**Comment by FBruzzesi** [comment]
@MarcoGorelli I will wait for an opinion before going rogue and implement all the queries 😂

**Comment by MarcoGorelli** [comment]
sure, looks good to me, thanks!

**Comment by DeaMariaLeon** [comment]
Should this be closed?

**Comment by FBruzzesi** [comment]
Hey @DeaMariaLeon thanks for the ping!

Sure I will close this as it is out of sync and wait for #863 to be finished before putting more effort into it

**Pull request closed without merging**